### ISSUE TITLE
There is a bug in the maintenance mode.

### ISSUE THREAD

**Issue by ZhiweiWu0425** [rainlab/pages-plugin#495]
在维护模式下有bug，开启维护模式，页面并不正常显示维护页面，会吧当前url的static-pages信息也显示出来

There is a bug in the maintenance mode. When the maintenance mode is turned on, the page does not display the maintenance page normally, so the static-pages information of the current url is also displayed

![image](https://user-images.githubusercontent.com/89636889/135076476-6b4892b2-0476-4ef3-85a8-2ec01e6a998c.png)
![dd](https://user-images.githubusercontent.com/89636889/135076868-623fecd9-ca51-4733-9d16-c3a1e7754f01.png)

**Comment by daftspunk** [comment]
Hi @ZhiweiWu0425

We can confirm that the maintenance mode is working with pages generated using this pages plugin. Note, if you are signed in as an administrator, then maintenance mode will not have any effect. Administrators can navigate the website when maintenance mode is turned on.

If this is still an issue, please provide step-by-step replication instructions on how to arrive at the problem.

I hope this helps.

**Comment by ZhiweiWu0425** [comment]
I understand what you said
But if the maintenance mode is enabled, ExtraData should not be added to the interface, and the maintenance interface should only display Pages/xx.htm set by the maintenance mode
I tried to add some code in RainLab\Pages\Components\StaticPage, which achieved the effect I said
I wonder if you understand what I mean

```php
/**
     * Find foreign view bag values and add them to
     * the component and page vars.
     */
    protected function defineExtraData()
    {

        /*
         * Maintenance mode
         */
        if (
            MaintenanceSetting::isConfigured() &&
            MaintenanceSetting::get('is_enabled', false) &&
            !BackendAuth::getUser()
        ) {
            /*
             * If the maintenance mode is enabled, ExtraData should not be added to the interface, 
             * and the maintenance interface should only display Pages/xx.htm
             */
            return true;
        }

        ....

        return $extraData;
    }
```
original
您说的我都明白
但是如果开启维护模式，不应吧ExtraData添加到界面，维护界面只应显示 维护模式设置的Pages/xx.htm
我在RainLab\Pages\Components\StaticPage中尝试加入了一些代码，达到我说的效果了
不知您是否理解我表达的意思

**Comment by daftspunk** [comment]
@ZhiweiWu0425

Got it! Fix can be found here 41474b756f36932dd319bbcdb07f6df8d8c13c70

Thanks

**Comment by ZhiweiWu0425** [comment]
Thank you for handling this issue quickly, everything is normal now, great! !
